File: Pidev2-3A-user (4)/Pidev2-3A-user/Pidev2-3A-user/src/Entity/Commande.php
<?php

namespace App\Entity;
use Symfony\Component\Validator\Constraints as Assert;

use Doctrine\ORM\Mapping as ORM;

/**
 * Commande
 *
 * @ORM\Table(name="commande", indexes={@ORM\Index(name="iduser", columns={"iduser"})})
 * @ORM\Entity
 */
class Commande
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_commande", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idCommande;

    /**
     * @var int
     *
     * @ORM\Column(name="prix_totale", type="integer", nullable=false)
     */
    #[Assert\NotBlank (message:"veuillez saisir prix totale ")]

    private $prixTotale;

    /**
     * @var string
     *
     * @ORM\Column(name="produits", type="string", length=250, nullable=false)
     */
    //#[Assert\NotBlank (message:"veuillez saisir produits ")]

    private $produits;

    /**
     * @var string
     *
     * @ORM\Column(name="adresse", type="string", length=250, nullable=false)
     */
    #[Assert\NotBlank (message:"veuillez saisir adresses correcte! ")]

    private $adresse;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="IdUser", referencedColumnName="IdUser")
     * })
     */    
    //#[Assert\NotBlank (message:"veuillez saisir id client ")]

    private $iduser;
    

    public function getIdCommande(): ?int
    {
        return $this->idCommande;
    }

    public function getPrixTotale(): ?int
    {
        return $this->prixTotale;
    }

    public function setPrixTotale(int $prixTotale): self
    {
        $this->prixTotale = $prixTotale;

        return $this;
    }

    public function getProduits(): ?string
    {
        return $this->produits;
    }

    public function setProduits(string $produits): self
    {
        $this->produits = $produits;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getIduser(): ?User
    {
        return $this->iduser;
    }

    public function setIduser(?User $iduser): self
    {
        $this->iduser = $iduser;

        return $this;
    }


}

File: src/com/mikhasaeb/entities/Commande.java
package com.mikhasaeb.entities;

import com.mikhasaeb.utils.Statics;

public class Commande implements Comparable<Commande> {

    private int id;
    private User user;
    private int prixTotale;
    private String produits;
    private String adresse;

    public Commande() {
    }

    public Commande(int id, User user, int prixTotale, String produits, String adresse) {
        this.id = id;
        this.user = user;
        this.prixTotale = prixTotale;
        this.produits = produits;
        this.adresse = adresse;
    }

    public Commande(User user, int prixTotale, String produits, String adresse) {
        this.user = user;
        this.prixTotale = prixTotale;
        this.produits = produits;
        this.adresse = adresse;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public int getPrixTotale() {
        return prixTotale;
    }

    public void setPrixTotale(int prixTotale) {
        this.prixTotale = prixTotale;
    }

    public String getProduits() {
        return produits;
    }

    public void setProduits(String produits) {
        this.produits = produits;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }


    @Override
    public int compareTo(Commande commande) {
        switch (Statics.compareVar) {
            case "User":
                return this.getUser().getEmail().compareTo(commande.getUser().getEmail());
            case "PrixTotale":
                return Integer.compare(this.getPrixTotale(), commande.getPrixTotale());
            case "Produits":
                return this.getProduits().compareTo(commande.getProduits());
            case "Adresse":
                return this.getAdresse().compareTo(commande.getAdresse());

            default:
                return 0;
        }
    }

}